### DISCUSSION TITLE
ScrollToIndex with smooth and Dynamic height wrong position

### DISCUSSION THREAD

**Discussion by jvinhit** [TanStack/virtual#536] [General] [answered]
I have an issue when using **scrollToIndex** with **behavior: smooth**. It's not working on large list data ( wrong position after scrolling).
If I removed behavior:smooth, it works well. I use version beta v54 and v30.
I found that code:
` if (behavior === 'smooth' && this.isDynamicMode()) {...`

I would like to ask a question, Can I manually control scrollToIndex smooth and without more jumps if custom this function?  If scrollIndex is the end of a large list => multiple jumps. 
```
const scrollToIndex = (index, ...args) => {
  rowVirtualizer.scrollToIndex(index, ...args);

  setTimeout(() => {
    const { start, end } = rangeRef.current;

    const isVisible = index >= start && index <= end;

    if (!isVisible) {
      scrollToIndex(index, ...args);
    }
  }, 1000);
};
```
thanks

**Comment by piecyk** [comment, marked as answer]
Hi, checkout this comment https://github.com/TanStack/virtual/discussions/495#discussioncomment-4886406
